include/exadg/incompressible_navier_stokes/time_integration/time_int_bdf_pressure_correction.cpp

```diff
@@ -318,19 +318,26 @@ void
 TimeIntBDFPressureCorrection<dim, Number>::do_timestep_solve()
 {
   // perform the sub-steps of the pressure-correction scheme
+  if(this->update_velocity)
+  {
+    momentum_step();
+  }
 
-  momentum_step();
-
-  VectorType pressure_increment;
-  pressure_increment.reinit(pressure_np, false /* init with zero */);
+  if(this->update_pressure)
+  {
+    VectorType pressure_increment;
+    pressure_increment.reinit(pressure_np, false /* init with zero */);
 
-  pressure_step(pressure_increment);
+    pressure_step(pressure_increment);
 
-  projection_step(pressure_increment);
+    projection_step(pressure_increment);
+  }
```

Comment:
I have solved this issue now reusing the anyways stored `pressure_increment_last_iter`.
One could also recompute the pressure increment (using a `+div(u)` term for the rotational schemes, so its not a simple vector addition unfortunately).
But since recovering velocity and pressure is only relevant for multiphysics problems (via `advance_one_timestep_partitioned_solve()`), where we always store the solution, we might as well use it :smile: